R/get_employee_file.R

```diff
@@ -1,22 +1,49 @@
-#' @title Bamboo HR API get request wrapper
+#' @title Retrieve An Employee File
 #'
-#' @description  Submits get requests to the Bamboo API
+#' @description `get_employee_file` takes 'id' (string), 'file_id', (string), and
+#' 'api_version' (string) and then requests and returns data about
+#' the corresponding file from the BambooHR API.
+#' The file will then be written to the user's working directory if possible.
 #'
-#' @param id employee ID. The special employee ID of zero (0) means to use the employee ID associated with the API key (if any).
-#' @param file_id  vector of values
-#' @param verbose Logical scalar. Should the function provide verbose messaging back on each step? - (XL) Sounds good
-#' @param ... tbd
+#' @param id The employee id of the employee.
+#' @param file_id The file id of the file to be returned.
+#' @param api_version The version of BambooHR API to be used.
 #'
-#' @return response object
-#' @author Mark Druffel, \email{mdruffel@propellerpdx.com}
-get_employee_file <- function(id, file_id, verbose, ...){
-  dots <- rlang::list2(...)
-  #Default to view if a file_id is not specified, this provides all files the employee has
+#' @return returns a response object.
+#'
+#' @examples
+#' response <- get_employee_file(
+#' id = 0,
+#' file_id = "480",
+#' api_version = "v1"
+#' )
+#'
+#' @author Harry Alexander, \email{harry.alexander@ascent.io}
+#'
+get_employee_file <- function(id, file_id,
+                              api_version = "v1") {
+  url <- build_url(api_version = api_version)
+  # default to 0 if employee id is not specified
+  id <- rlang::maybe_missing(id, default = "0")
```

Comment:
tried the function with id not specified, or specified as either "0" or 0, and all 3 returned an error (403), providing id explicitly - i.e. using my employee id does work though